src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -1,647 +1,560 @@
 package com.minecolonies.entity.ai.citizen.farmer;
 
+import com.minecolonies.blocks.BlockHutField;
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.Field;
 import com.minecolonies.colony.buildings.BuildingFarmer;
 import com.minecolonies.colony.jobs.JobFarmer;
+import com.minecolonies.entity.EntityCitizen;
 import com.minecolonies.entity.ai.basic.AbstractEntityAIInteract;
-import com.minecolonies.util.BlockPosUtil;
+import com.minecolonies.entity.ai.util.AIState;
+import com.minecolonies.entity.ai.util.AITarget;
+import com.minecolonies.util.BlockUtils;
 import com.minecolonies.util.InventoryUtils;
-import net.minecraft.block.Block;
+import net.minecraft.block.BlockCarrot;
 import net.minecraft.block.BlockCrops;
+import net.minecraft.block.IGrowable;
+import net.minecraft.block.state.IBlockState;
 import net.minecraft.init.Blocks;
-import net.minecraft.init.Items;
 import net.minecraft.item.Item;
+import net.minecraft.item.ItemSeeds;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.BlockPos;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import net.minecraftforge.common.IPlantable;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.concurrent.CopyOnWriteArrayList;
+import static com.minecolonies.colony.Field.FieldStage.*;
+import static com.minecolonies.entity.ai.util.AIState.*;
 
 /**
  * Farmer AI class
  * Created: December 20, 2014
- *
- * @author Raycoms, Kostronor
  */
-
 public class EntityAIWorkFarmer extends AbstractEntityAIInteract<JobFarmer>
 {
-    private static final String         TOOL_TYPE_HOE     = "hoe";
-    private static final String         TOOL_TYPE_SHOVEL  = "shovel";
-    private static final String         RENDER_META_SEEDS = "Seeds";
-    private static final Logger         logger            = LogManager.getLogger("Farmer");
-    private              List<BlockPos> farmAbleLand      = new ArrayList<>();
-    private              List<BlockPos> plowedLand        = new ArrayList<>();
-    private              List<BlockPos> crops             = new ArrayList<>();
-    private              List<BlockPos> crops2            = new ArrayList<>();
-    private BlockPos currentFarmLand;
-    private int    harvestCounter = 0;
-    private String needItem       = "";
-    private int delay = 0;
-
-    //TODO Check for duplicates
-    public EntityAIWorkFarmer(JobFarmer job)
-    {
-        super(job);
-    }
-    //TODO Planting randomly depending on option in Hut, each level, one more crop type
-    //TODO Adding Language Strings in files
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)